crates/ruff_server/src/session/workspace/ruff_settings.rs

```diff
@@ -149,6 +150,24 @@ impl<'a> ConfigurationTransformer for EditorConfigurationTransformer<'a> {
             ..Default::default()
         };
 
+        if let Some(config_file_path) = configuration {
+            match open_configuration_file(&config_file_path, project_root) {
+                Ok(config_from_file) => {
+                    editor_configuration = editor_configuration.combine(config_from_file);
+                }
+                Err(err) => tracing::error!("Unable to find custom configuration file {err}"),
+            }
+        }
+
         editor_configuration.combine(project_configuration)
```

Comment:
If `configuration` is specified, should we ignore project configuration entirely? (My read is that we're merging them.) That's how `--config` works in the Ruff CLI.

Reply:
My plan was to merge them - if we wanted to allow for this behavior, we could introduce an additional resolution strategy that allows the user to ignore project configuration.

Reply:
I worry about adding a bunch of resolution strategies. They're hard to reason about -- even I'm having trouble keeping the rules straight.

I think merging here is surprising given that the Ruff CLI does _not_ do this -- if you provide a configuration file, that's the configuration you get.

Reply:
What's your current thinking here @snowsignal?

Reply:
I've been deliberating over this, and I actually do think it makes sense to have this reflect the behavior of `--config`. Having three different groups of options merging into each other is a lot of cognitive complexity, and a full override does feel like a more 'intuitive' option.

My assumption is that if we were using a `prioritizeWorkspace` resolution strategy, this configuration file would be ignored.